src/spikeinterface/sorters/utils/misc.py

```diff
@@ -62,12 +62,16 @@ def has_nvidia():
     """
     Checks if the machine has nvidia capability.
     """
-    from cuda import cuda
+    
+    try:
+        from cuda import cuda
+    except ModuleNotFoundError as err:
+        raise Exception("This container requires cuda, but the package is not installed. You can install it with:\npip install cuda-python") from err
```

Comment:
This is strange because `cuda-python` is included in `full` (see https://github.com/SpikeInterface/spikeinterface/blob/main/pyproject.toml#L93) and we install `[full]` in the container: https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/sorters/runsorter.py#L494-L519

I'm assuming you're using a custom image then @DradeAW ?

Reply:
I install SI using `pip install -e .` which only installs the core dependencies (and not the full). So I always install the required packages one by one if I see one is missing (probably not the best way to do it, but oh well).

For the image I'm using the regular one for `kilosort3`.

Reply:
What do you mean?

I'm in my standard conda environment, and this env is missing `cuda-python`, so the line `ss.run_sorter("kilosort3", singularity_image=True)` crashes.